packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,284 @@
+use pathdiff::diff_paths;
+use regex::{Captures, Regex};
+use relay_compiler_common::SourceLocationKey;
+use std::borrow::Cow;
+use std::path::PathBuf;
+
+use relay_compiler::compiler_state::{SourceSet, SourceSetName};
+use relay_compiler::{create_path_for_artifact, FileCategorizer, FileGroup, ProjectConfig};
+use serde::Deserialize;
+use swc_atoms::JsWord;
+use swc_common::errors::HANDLER;
+use swc_common::FileName;
+use swc_ecmascript::ast::*;
+use swc_ecmascript::utils::{quote_ident, ExprFactory};
+use swc_ecmascript::visit::{Fold, FoldWith};
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayLanguageConfig {
+    Typescript,
+    Flow,
+}
+
+struct Relay {
+    file_name: FileName,
+    relay_config_for_tests: Option<ProjectConfig>,
+}
+
+fn pull_first_operation_name_from_tpl(tpl: &TaggedTpl) -> Option<String> {
+    tpl.tpl
+        .quasis
+        .iter()
+        .filter_map(|quasis| {
+            let regex = Regex::new(r"(fragment|mutation|query|subscription) (\w+)").unwrap();
```

Comment:
This will be very slow. Please use cache. 
You can use `static` with `once_cell::sync::Lazy`